src/main/java/com/espro/flink/consul/metric/ConsulMetricServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.espro.flink.consul.metric;
+
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.Metric;
+import org.apache.flink.metrics.SimpleCounter;
+import org.apache.flink.runtime.metrics.MetricRegistry;
+import org.apache.flink.runtime.metrics.DescriptiveStatisticsHistogram;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConsulMetricServiceImpl implements ConsulMetricService {
+    private static final Logger LOG = LoggerFactory.getLogger(ConsulMetricServiceImpl.class);
+
+    private static final int HISTORY_SIZE = 100;
+
+    private final MetricRegistry metricRegistry;
+    private final ConsulMetricGroup consulMetricGroup;
+    private final Counter counter;
+    private final Histogram histogram;
+    private final Map<String, Metric> metricMap = new HashMap<>();
+
+    public ConsulMetricServiceImpl(MetricRegistry metricRegistry, ConsulMetricGroup consulMetricGroup) {
+        this.metricRegistry = metricRegistry;
+        this.consulMetricGroup = consulMetricGroup;
+        this.counter = new SimpleCounter();
+        this.histogram = new DescriptiveStatisticsHistogram(HISTORY_SIZE);
+        initDefaultMetrics();
+    }
+
+    @Override
+    public void registerDefaultMetrics() {
+        LOG.info("Start registering default consul metrics.");
+        this.metricMap.forEach((metricName, metric) -> registerMetrics(metric, metricName));
```

Comment:
Looks complicated an error prone to create a map as instance variable just for using this map a single time to build another data structure. Why not simply write down the registrations "by hand"?